datafusion/core/src/execution/context.rs

```diff
@@ -1897,6 +1874,35 @@ impl FunctionRegistry for SessionState {
     }
 }
 
+impl OptimizerConfig for SessionState {
+    fn query_execution_start_time(&self) -> DateTime<Utc> {
+        self.execution_props.query_execution_start_time
+    }
+
+    fn rule_enabled(&self, name: &str) -> bool {
```

Comment:
I eventually hope to move `ConfigOptions` into core, which will allow simplifying this